fl4health/model_bases/pca.py

```diff
@@ -0,0 +1,166 @@
+from logging import INFO
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from torch import Tensor
+from torch.nn.parameter import Parameter
+
+
+class PCAModule(nn.Module):
+    def __init__(self, low_rank: bool = True, full_svd: bool = False, rank_estimation: int = 6) -> None:
+        """
+        PyTorch module for performing Principal Component Analysis.
+
+        Args:
+            low_rank (bool, optional): Indicates whether the data matrix is low-rank. If the user has
```

Comment:
Is this approximately low rank or exactly low rank. That is, does the matrix have to have an analytical low rank representation? I imagine it is "approximately low rank" where this means that the matrix can be well-approximated with a significantly smaller set of principal components than the full rank, but it would be good to be explicit for the user.

Additionally, what is the cost to getting this wrong. That is, I think the matrix is of low approximate rank, but it turns out it isn't. That's worth mentioned I think?

Reply:
**Note**: You can also just link to the SVD implementation documentation if you think they do a good job explaining such things 🙂

Reply:
I think these are important questions. I've added more comments addressing them, and we can discuss whether the updates have cleared things up 🙂